documentation/guide/writing-persistent-objects.rst

```diff
@@ -0,0 +1,568 @@
+==========================
+Writing persistent objects
+==========================
+
+In the :ref:`Tutorial <tutorial-label>`, we discussed the basics of
+implementing persistent objects by subclassing
+``persistent.Persistent``.  This is probably enough for 80% of
+persistent-object classes you write, but there are some other aspects
+of writing persistent classes you should be aware of.
+
+Access and modification
+=======================
+
+Two of the main jobs of the ``Persistent`` base class is to detect
+when an object has been accessed and when it has been modified.  When
+an object is accessed, its state may need to be loaded from the
+database.  When an object is modified, the modification needs to be
+saved if a transaction is committed.
+
+``Persistent`` detects object accesses by hooking into object
+attribute access and update.  In the case of object update, there
+may be other ways of modifying state that we need to make provision for.
+
+Rules of persistence
+====================
+
+When implementing persistent objects, be aware that an object's
+attributes should be :
+
+- immutable (such as strings or integers),
+
+- persistent (subclass Persistent), or
+
+- You need to take special precautions.
+
+If you modify a non-persistent mutable value of a persistent-object
+attribute, you need to mark the persistent object as changed yourself
+by setting ``_p_changed`` to True::
```

Comment:
I don't understand this comment.  You seem to be saying that reading _p_changed doesn't give an accurate result.

_p_changed is computed from and thus reflects the internal object state. Can you provide an example where reading _p_changed doesn't give the expected result?